sys/action.js

```diff
@@ -137,7 +137,14 @@ function buildQueryResponse(res) {
     } else if (!res.body || res.body.error) {
         throw apiError((res.body || {}).error);
     } else if (!res.body.query || (!res.body.query.pages && !res.body.query.userinfo)) {
-        throw apiError({ info: 'Missing query pages from the PHP action API response.' });
+        throw new HTTPError({
+            status: 404,
+            body: {
+                type: 'not_found#page_revisions',
```

Comment:
This looks wrong. The query module has *a lot* of sub-modules, so emitting revinfo 404s here doesn't look good to me.